doc/doxygen/src/advanced-build-system-tricks.md

```diff
@@ -221,15 +221,57 @@ the debugger hardware. With the `make list-ttys` it is reported as the 'serial':
-/sys/bus/usb/devices/1-1.4.3: SEGGER J-Link, serial: '000683806234', tty(s): ttyACM0
+Path          | Driver     | Vendor               | Model                          | Serial                         | ctime
+--------------|------------|----------------------|--------------------------------|--------------------------------|---------
+/dev/ttyUSB0  | cp210x     | Silicon Labs         | CP210x UART Bridge             | 0001                           | 13:28:12
+/dev/ttyACM1  | cdc_acm    | STMicroelectronics   | ST-LINK/V2.1                   | 0670FF485151717867105121       | 13:34:17
+/dev/ttyACM2  | cdc_acm    | SEGGER               | J-Link                         | 000683475134                   | 13:46:25
+/dev/ttyACM0  | cdc_acm    | LG Electronics US... | 100 Series/C230 Series Chip... | ABC123456789                   | 08:31:46
 ~~~~~~~~~~~~~~~~~~~
 
 When the above make snippet is included as `RIOT_MAKEFILES_GLOBAL_PRE`, the
 serial number of the USB device is automatically set if the used board is
 included in the script. This will then ensure that the board debugger is used
 for flashing and the board serial device is used when starting the serial console.
 
+Alternative Tool: ttys.py
+-------------------------
+
+As an alternative to above shell script, `$(RIOTTOOLS)/usb-serial/ttys.py` can
+be used. If called without arguments, it will print all USB attached TTYs in
+a table, e.g. like this:
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Path          | Driver     | Vendor               | Model                          | Serial                         | ctime
+--------------|------------|----------------------|--------------------------------|--------------------------------|---------
+/dev/ttyUSB0  | cp210x     | Silicon Labs         | CP210x UART Bridge             | 0001                           | 13:28:12
+/dev/ttyACM1  | cdc_acm    | STMicroelectronics   | ST-LINK/V2.1                   | 0670FF485151717867105121       | 13:34:17
+/dev/ttyACM2  | cdc_acm    | SEGGER               | J-Link                         | 000683475134                   | 13:46:25
+/dev/ttyACM0  | cdc_acm    | LG Electronics US... | 100 Series/C230 Series Chip... | ABC123456789                   | 08:31:46
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+It supports command line parameters to filter by vendor name, model name, serial
+number, or driver. In addition, the `--most-recent` argument will only print the
+most recently added interface (out of those matching the filtering by vendor,
+model, etc.). The `--path-only` argument will result in only the device path
+being printed for convenient use in scripts.
+
+The following Make snippet can be use to connect to the most recently connected
+board:
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~{Makefile}
+PORT ?= $(shell $(RIOTTOOLS)/usb-serial/ttys.py --most-recent --format path)
```

Comment:
Does it still make sense to have that snippet around now that MOST_RECENT_PORT is available?

Reply:
I will try to come up with a more sensible example.